controller/src/beerocks/master/tasks/task.cpp

```diff
@@ -33,11 +33,14 @@ void task::response_received(std::string mac, beerocks_message::eActionOp_CONTRO
         if ((it->second == action_op) && (it->first == mac)) {
             it = pending_macs.erase(it);
             handle_response(mac, action_op, cmdu_rx);
-            break;
+            return;
         } else {
             it++;
         }
     }
+
+    // Handle the response even if we are not expecting it
+    handle_response(mac, action_op, cmdu_rx);
```

Comment:
This looks like a real hack and I'm not sure it's worth it. Why did we do this change in the demo? At least we should add a print when such a thing happens, but I rather drop this commit altogether.

Reply:
It is an optimization for the optimal path task.
Imagine the optimal path task starting to ask a measurement on each BSSID at the time.
Now, let's assume that the task requested measurement on BSSID_A and waited for a response, but reached timeout, and then started to ask for measurement on BSSID_B, but while doing so, received the response for BSSID_A. If it is good, it will be used, and will save an additional measurement to BSSID_A.
If you take into account that each measurement waits for a response to at most 6 seconds and that the second measurement might be bad (when you already received a good measurement), you have a lot to lose here.

Reply:
still feels like a hack, but OK I guess..